src/snowcli/cli/streamlit/manager.py

```diff
@@ -47,9 +88,21 @@ def deploy(
         stage_name: Optional[str] = None,
         warehouse: Optional[str] = None,
         replace: Optional[bool] = False,
+        experimental: Optional[bool] = False,
     ):
         stage_manager = StageManager()
 
+        if experimental:
+            return self._deploy_experimental(
+                streamlit_name,
+                main_file,
+                environment_file,
+                pages_dir,
+                stage_name,
+                warehouse,
+                replace,
+            )
+
```

Comment:
I would say that this part should be in command implemtation so we can explicitly see what parts differs

Reply:
Yeah, that's what I originally sought to do, but the challenge is that the order is inherently different -- in the normal case you create the stage, upload to it, and then create the streamlit. In the experimental version, you create the streamlit then upload to the embedded stage. I suppose if I refactor out some pieces I could simplify it down to:

```
if experimental:
   create_streamlit()
   upload_stuff()
else:
   create_stage()
   upload_stuff()
   create_streamlit()
```

Do you feel like that would be clearer?

Reply:
I refactored a bit to add more shared functionality between the different deploy methods